docs/large_inputs.md

```diff
@@ -1,63 +1,127 @@
+
+By default, Dataflow uses its autoscaling algorithm to adjust the number of
+workers assigned to each job. You may adjust the maximum number of workers using
+`--max_num_workers`. You may also use `--num_workers` to specify the initial
+number of workers to assign to the job.
+
+## `--worker_machine_type`
 
 By default, Dataflow uses the `n1-standard-1` machine, which has 1 vCPU and
-3.75GB of RAM. It is recommanded to use a larger machine type for large
-datasets. You may change the machine type by adding
-`--worker_machine_type <machine type>` to the pipeline args. Please see
-https://cloud.google.com/compute/pricing#predefined_machine_types for a list of
-available machine types.
+3.75GB of RAM. You may need to request a larger machine for large datasets.
+Please see https://cloud.google.com/compute/pricing#predefined_machine_types
+for a list of available machine types.
+
+We have observed that Dataflow performs more efficiently when running
+with a large number of small machines rather than a small number of large
+machines (e.g. 200 `n1-standard-4` workers instead of 25 `n1-standard-32`
+workers). This is due to disk/network IOPS (input/output operations per second)
+being limited for each machine especially if [merging](variant_merging.md) is
+enabled.
 
-Please note that you may need to adjust your quota to accomoddate the additional
-number of CPUs.
+Using a large number of workers may not always be possible due to disk and
+IP quotas. As a result, we recommend using SSDs (see
+[`--worker_disk_type`](#--worker_disk_type)) when choosing a large (more than
+`n1-standard-8`) machine, which yields higher disk IOPS and can avoid idle CPU
+cycles. Note that disk is significantly cheaper than CPU, so always try to
+optimize for high CPU utilization rather than disk usage.
 
-## Disk size
+## `--disk_size_gb`
 
 By default, each worker has 250GB of disk. The aggregate amount of disk space
 from all workers must be at least as large as the uncompressed size of all VCF
 files being processed. However, to accomoddate for intermediate stages of the
 pipeline and also to account for the additional overhead introduced by the
 transforms (e.g. the sample name is repeated in every record in the BigQuery
 output rather than just being specified once as in the VCF header), you
-typically need 3 to 4 times the total size of the raw VCF files. You may change
-the disk size of every worker by adding `--disk_size_gb <disk size>` to the
-pipeline args.
+typically need 3 to 4 times the total size of the raw VCF files.
+
+In addition, if [merging](variant_merging.md) or
+[--num_bigquery_write_shards](#--num_bigquery_write_shards) is enabled, you may
+need more disk per worker (e.g. 500GB) as the same variants need to be
+aggregated together on one machine.
+
+## `--worker_disk_type`
+
+SSDs provide significantly more IOPS than standard persistent disks, but are
+more expensive. However, when choosing a large machine (e.g. `n1-standard-16`),
```

Comment:
You referred to a large machine earlier as `n1-standard-8`, I think it's better to be consistent.

Reply:
ah, that one was "larger than" 8 :p rephrased... :)